ObjectiveC: support cateogries

masatake opened the pull request:
About category of Objective-C, see https://medium.com/ios-os-x-development/an-introduction-to-categories-in-objective-c-d6d55d91569f

Close #1487

b4n commented on parsers/objc.c:
```diff
@@ -126,6 +131,24 @@ static const keywordTable objcKeywordTable[] = {
+static fieldDefinition ObjcFields [] = {
+	{
+		.name = "category",
+		.description = "category attached to th class",
```
*to th**e** class* (e is missing)

masatake replied:
Thank you. I will update in soon.

masatake commented:
After thinking again, defining two kinds, categoryIface and categoryImpl looks overkill.
I will revise this.

masatake commented:
Updated.
categoryImpl and categoryIface kinds are removed. Instead, I introduce category kind.
We can known whether a category is defined in @interface  line or @implementation line from scope information. So we don't have to introduce the two kinds.
Using scope field to know whether a category is defined in @interface  line or @implementation line is
a bit abuse. However, I think it is better than introducing the two kinds.

In addition, I introduced code for parsing protocols spec in @interface line.
